src/installer.ts

```diff
@@ -193,111 +162,107 @@ export class DotnetCoreInstaller {
         );
       }
     }
+  }
 
-    console.log(process.env['PATH']);
+  constructor(version: string, quality: string) {
+    this.version = version;
+    this.quality = quality;
   }
 
-  // versionInfo - versionInfo of the SDK/Runtime
-  async resolveVersion(versionInfo: DotNetVersionInfo): Promise<string> {
-    if (versionInfo.isExactVersion()) {
-      return versionInfo.version();
+  private setQuality(
+    dotnetVersion: IDotNetVersion,
+    scriptArguments: string[]
+  ): void {
+    const option = IS_WINDOWS ? '-Quality' : '--quality';
+    if (dotnetVersion.qualityFlag) {
+      scriptArguments.push(option, this.quality);
+    } else {
+      logWarning(
+        `'dotnet-quality' input can be used only with .NET SDK version in A.B, A.B.x, A and A.x formats where the major tag is higher than 5. You specified: ${this.version}. 'dotnet-quality' input is ignored.`
+      );
     }
+  }
 
-    const httpClient = new hc.HttpClient('actions/setup-dotnet', [], {
-      allowRetries: true,
-      maxRetries: 3
-    });
-
-    const releasesJsonUrl: string = await this.getReleasesJsonUrl(
-      httpClient,
-      versionInfo.version().split('.')
-    );
-
-    const releasesResponse = await httpClient.getJson<any>(releasesJsonUrl);
-    const releasesResult = releasesResponse.result || {};
-    let releasesInfo: any[] = releasesResult['releases'];
-    releasesInfo = releasesInfo.filter((releaseInfo: any) => {
-      return (
-        semver.satisfies(releaseInfo['sdk']['version'], versionInfo.version(), {
-          includePrerelease: this.includePrerelease
-        }) ||
-        semver.satisfies(
-          releaseInfo['sdk']['version-display'],
-          versionInfo.version(),
-          {
-            includePrerelease: this.includePrerelease
-          }
-        )
-      );
-    });
+  public async installDotnet() {
+    const windowsDefaultOptions = [
+      '-NoLogo',
+      '-Sta',
+      '-NoProfile',
+      '-NonInteractive',
+      '-ExecutionPolicy',
+      'Unrestricted',
+      '-Command'
+    ];
+    const scriptName = IS_WINDOWS ? 'install-dotnet.ps1' : 'install-dotnet.sh';
+    const escapedScript = path
+      .join(__dirname, '..', 'externals', scriptName)
+      .replace(/'/g, "''");
+    let scriptArguments: string[];
+    let scriptPath = '';
+
+    const versionResolver = new DotnetVersionResolver(this.version);
+    const dotnetVersion = await versionResolver.createDotNetVersion();
+
+    const envVariables: {[key: string]: string} = {};
+    for (let key in process.env) {
+      if (process.env[key]) {

```

Comment:
Looks like this `for` loop just filters out non-falsy keys from `process.env`.  Why do we need to do that?